pkg/util/clusterdata/cluster_version_task.go

```diff
@@ -0,0 +1,55 @@
+package clusterdata
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+func newClusterVersionEnricherTask(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftCluster) (enricherTask, error) {
+	client, err := configclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &clusterVersionEnricherTask{
+		log:    log,
+		client: client,
+		oc:     oc,
+	}, nil
+}
+
+type clusterVersionEnricherTask struct {
+	log    *logrus.Entry
+	client configclient.Interface
+	oc     *api.OpenShiftCluster
+}
+
+func (ef *clusterVersionEnricherTask) FetchData(callbacks chan<- func()) {
+	cv, err := ef.client.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+	if err != nil {
+		ef.log.Error(err)
+		callbacks <- func() {}
+		return
+	}
+
+	if cv.Status.Desired.Version == "" {
+		ef.log.Error("cluster version not found")
+		callbacks <- func() {}
+		return
+	}
+
+	callbacks <- func() {
+		ef.oc.Properties.ClusterProfile.Version = cv.Status.Desired.Version
+	}
+}
+
+func (ef *clusterVersionEnricherTask) SetDefaults() {
+	ef.oc.Properties.ClusterProfile.Version = "UNKNOWN"
```

Comment:
I am still wondering whether the zero value would be better than "UNKNOWN", -1, etc

Reply:
If we return the zero value, it manifests as a missing json field, rather than as something that we would invalidate if it was sent back to us.  Seems less badly behaved.

Reply:
I decided to replace dumb values with zero values for now. Let me know if you see reasons to revert it.